File: src/components/TerminalsList.ts
// TerminalsList Component
// Generates a list of child terminals for a parent stop

/**
 * Generates a list of terminals
 */
export function TerminalsList(childStopIds: string[], childStopNames: string[]): string {
  if (childStopIds.length === 0) {
    return '';
  }
  
  let html = `\n  <h2>Terminals</h2>\n`;
  html += `  <ul>\n`;
  
  for (let i = 0; i < childStopIds.length; i++) {
    html += `    <li><a href="/stops/${childStopIds[i]}/index.html">${childStopNames[i]}</a></li>\n`;
  }
  
  html += `  </ul>\n`;
  
  return html;
}

File: rust_src/components/terminals_list.rs
// Terminals List Component
use crate::modules::models::ChildStop;

pub fn render(child_stops: &[ChildStop], _agency_id: &str, parent_stop_id: &str) -> String {
    if child_stops.is_empty() {
        return String::new();
    }
    
    let mut html = String::from("  <h2>Terminals</h2>\n  <ul>\n");
    
    for child in child_stops {
        html.push_str(&format!(
            "    <li><a href=\"/stops/{}/{}/\">{}</a></li>\n",
            parent_stop_id, child.id, child.name
        ));
    }
    
    html.push_str("  </ul>\n");
    html
}
